src/js/providers/video-actions-mixin.js

```diff
@@ -18,23 +20,28 @@ const VideoActionsMixin = {
     },
 
     resize: function(width, height, stretching) {
-        if (!width || !height || !this.video.videoWidth || !this.video.videoHeight) {
+        const fitVideoUsingTransforms = Browser.ie || (OS.iOS && OS.version.major < 9) || Browser.androidNative;        
+        if ((!width || !height || !this.video.videoWidth || !this.video.videoHeight) && !fitVideoUsingTransforms) {
             return false;
         }
+        const styles = {
+            objectFit: '',
+            width: '',
+            height: '',
+        };
         if (stretching === 'uniform') {
             // snap video to edges when the difference in aspect ratio is less than 9%
             var playerAspectRatio = width / height;
             var videoAspectRatio = this.video.videoWidth / this.video.videoHeight;
-            var objectFit = null;
             if (Math.abs(playerAspectRatio - videoAspectRatio) < 0.09) {
-                objectFit = 'fill';
+                styles.objectFit = 'fill';
+                stretching = 'exactfit';
             }
-            style(this.video, {
-                objectFit,
-                width: null,
-                height: null
-            });
         }
+        if (fitVideoUsingTransforms && !this.video.videoWidth < width && !this.video.videoHeight < height) {
```

Comment:
These conditions are not logically correct. You're converting `videoWidth` and `videoHeight` to a boolean, then comparing with a number. What is the objective here?

Reply:
The bang operator was a mistake; but this essentially is designed to achieve two things:

(1) is the user in an environment that requires the fitToBounds method, as opposed to one that can utilize object-fit
(2) Ensure that the video should be fit to the bounds of the player -- in an HLS/Dash stream, if we up switch quality, then the height/width of the tag can exceed the player bounds. These two additional checks ensure that the video remains within the player bounds.